### ISSUE TITLE
SyncOnWifiOnly causes OfflineException on several actions

### ISSUE THREAD

**Issue by stefan-niedermann** [stefan-niedermann/nextcloud-deck#165]
**To Reproduce**
1. Set `Sync only on wifi` to true in the settings
2. Disable wifi
3. assign a label or a user to a card or unassign it

**Expected behavior**
`OfflineException` should be catched and the changes (assigns/unassigns) should be made local and synced next time when internet (= wifi in this case) is available again.

**Stacktrace**
```
it.niedermann.nextcloud.deck.exceptions.OfflineException: Device is currently offline.
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.ServerAdapter.ensureInternetConnection(ServerAdapter.java:77)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.ServerAdapter.assignUserToCard(ServerAdapter.java:205)
	at it.niedermann.nextcloud.deck.persistence.sync.SyncManager.lambda$assignUserToCard$20$SyncManager(SyncManager.java:527)
	at it.niedermann.nextcloud.deck.persistence.sync.-$$Lambda$SyncManager$5F76_PvwO1JZLCAN9OfGCfMpLA4.run(Unknown Source:6)
	at java.lang.Thread.run(Thread.java:764)
```

**Comment by desperateCoder** [comment]
Implemented a offline-check. If there is nothing more to do (please test it), fell free to close.

**Comment by stefan-niedermann** [comment]
Given `syncOnlyOnWifi` ist `true` and wifi is disabled.
When you unassign a user from a card:

```
2019-10-12 18:36:14.286 26984-27139/? E/AndroidRuntime: FATAL EXCEPTION: Thread-37
    Process: it.niedermann.nextcloud.deck, PID: 26984
    it.niedermann.nextcloud.deck.exceptions.OfflineException: Device is currently offline.
        at it.niedermann.nextcloud.deck.persistence.sync.adapters.ServerAdapter.ensureInternetConnection(ServerAdapter.java:77)
        at it.niedermann.nextcloud.deck.persistence.sync.adapters.ServerAdapter.unassignUserFromCard(ServerAdapter.java:210)
        at it.niedermann.nextcloud.deck.persistence.sync.SyncManager.lambda$unassignUserFromCard$23$SyncManager(SyncManager.java:584)
        at it.niedermann.nextcloud.deck.persistence.sync.-$$Lambda$SyncManager$FSBwmE6kRp1jZmT052ri7RbV_GE.run(Unknown Source:6)
        at java.lang.Thread.run(Thread.java:764)
```

**Comment by stefan-niedermann** [comment]
When opening About page with disabled Wi-Fi and `syncOnlyOnWifi: true`:

```
Version: 0.0.12

it.niedermann.nextcloud.deck.exceptions.OfflineException: Device is currently offline.
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.ServerAdapter.ensureInternetConnection(ServerAdapter.java:77)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.ServerAdapter.getCapabilities(ServerAdapter.java:133)
	at it.niedermann.nextcloud.deck.persistence.sync.SyncManager.getServerVersion(SyncManager.java:173)
	at it.niedermann.nextcloud.deck.ui.about.AboutFragmentCreditsTab.onCreateView(AboutFragmentCreditsTab.java:41)
	at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2600)
	at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:881)
	at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1238)
	at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1303)
	at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:439)
	at androidx.fragment.app.FragmentManagerImpl.executeOps(FragmentManagerImpl.java:2079)
	at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1869)
	at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1824)
	at androidx.fragment.app.FragmentManagerImpl.execSingleAction(FragmentManagerImpl.java:1696)
	at androidx.fragment.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:299)
	at androidx.fragment.app.FragmentPagerAdapter.finishUpdate(FragmentPagerAdapter.java:235)
	at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1244)
	at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1092)
	at androidx.viewpager.widget.ViewPager.onMeasure(ViewPager.java:1622)
	at android.view.View.measure(View.java:23190)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
	at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
	at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
	at android.view.View.measure(View.java:23190)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
	at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:143)
	at android.view.View.measure(View.java:23190)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
	at androidx.appcompat.widget.ActionBarOverlayLayout.onMeasure(ActionBarOverlayLayout.java:403)
	at android.view.View.measure(View.java:23190)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
	at android.view.View.measure(View.java:23190)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
	at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
	at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
	at android.view.View.measure(View.java:23190)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6749)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
	at com.android.internal.policy.DecorView.onMeasure(DecorView.java:716)
	at android.view.View.measure(View.java:23190)
	at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2725)
	at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1575)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1858)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1463)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7190)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
	at android.view.Choreographer.doCallbacks(Choreographer.java:761)
	at android.view.Choreographer.doFrame(Choreographer.java:696)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6718)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by stefan-niedermann** [comment]
But i think this might be a valid case to throw an OfflineException. This should be handeled on the UI side maybe.

**Comment by desperateCoder** [comment]
Same for fetching the activities, right? Currently I don't sync them when offline, but we should, right?

**Comment by stefan-niedermann** [comment]
Activities are fine right now - It doesn't blow up the application and we have nice text there which explains that activities are only loaded when online.

That's enough for now. I fixed the about activity in e5de3bfe71de50f19fd800d4df7e676c27112e54.
